Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius, drawColor);
+
+		blendAmount = 120 + RandomNum(0, 50);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius - 1, drawColor);
+	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::Draw(BITMAP *drawBitmap) const {
+		drawing_mode(DRAW_MODE_TRANS, nullptr, 0, 0);
+		DrawSitePoints(drawBitmap);
+		if (m_SelectedScene && m_ScenarioCollectionBoxes.at(ScenarioCollections::SceneInfoBox)->GetVisible()) { DrawLineToSitePoint(drawBitmap); }
+		drawing_mode(DRAW_MODE_SOLID, nullptr, 0, 0);
+
+		AllegroScreen drawScreen(drawBitmap);
+		m_ScenarioGUIController->Draw(&drawScreen);
+
+		// Draw scene preview after GUI
+		if (m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox)->GetVisible()) {
+			GUICollectionBox *playerSetupBox = m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox);
+
+			// Draw the Player-Team matrix lines and disabled overlay effects.
+			int linePosY = 80;
+			for (int teamIndex = Activity::TeamOne; teamIndex < Activity::MaxTeamCount; ++teamIndex) {
+				if (m_SelectedActivity && (!m_SelectedActivity->TeamActive(teamIndex) || m_LockedCPUTeam == teamIndex)) {
+					// Apply a colored overlay on top of team rows that are not human-playable.
+					drawing_mode(DRAW_MODE_TRANS, nullptr, 0, 0);
+					int blendAmount = 230;
+					set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+					rectfill(drawBitmap, playerSetupBox->GetXPos() + 110, playerSetupBox->GetYPos() + linePosY, playerSetupBox->GetXPos() + playerSetupBox->GetWidth() - 12, playerSetupBox->GetYPos() + linePosY + 25, c_GUIColorDarkBlue);
+					drawing_mode(DRAW_MODE_SOLID, nullptr, 0, 0);
+				}
+				// Cell border separator lines.
+				hline(drawBitmap, playerSetupBox->GetXPos() + 110, playerSetupBox->GetYPos() + linePosY, playerSetupBox->GetXPos() + playerSetupBox->GetWidth() - 12, c_GUIColorLightBlue);
+				linePosY += 25;
+			}
+
+			// Manually draw UI elements on top of colored rectangle.
+			for (int teamIndex = Activity::MaxTeamCount - 1; teamIndex >= Activity::TeamOne; --teamIndex) {
+				if (m_TeamTechSelect.at(teamIndex)->GetVisible()) {
+					m_TeamTechSelect.at(teamIndex)->Draw(&drawScreen);
+					if (m_TeamTechSelect.at(teamIndex)->IsDropped()) { m_TeamTechSelect.at(teamIndex)->GetListPanel()->Draw(&drawScreen); }
+				}
+				if (m_TeamAISkillSlider.at(teamIndex)->GetVisible()) {
+					m_TeamAISkillSlider.at(teamIndex)->Draw(&drawScreen);
+					m_TeamAISkillLabel.at(teamIndex)->Draw(&drawScreen);
+				}
+			}
+		}
+		m_ScenarioGUIController->DrawMouse();
+
+		int device = g_UInputMan.GetLastDeviceWhichControlledGUICursor();
+		BITMAP *deviceIcon = nullptr;
+
+		// Draw the active joystick's sprite next to the mouse.
+		if (device >= InputDevice::DEVICE_GAMEPAD_1) {
+			int mouseX = 0;
```

Comment:
Do you have to define these as 0, 0? I'd expect GetMousePosition to always fill them in?
Also, seems like it'd be good to trash the comments and pull this stuff into into a private `DrawControllerMouseIcon` submethod or something